datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/dataset/mappers/ForeignKeyConstraintMapper.java

```diff
@@ -34,7 +37,12 @@ public static ForeignKeyConstraint map(com.linkedin.schema.ForeignKeyConstraint
 
   private static SchemaFieldEntity mapSchemaFieldEntity(Urn schemaFieldUrn) {
     SchemaFieldEntity result = new SchemaFieldEntity();
-    result.setParent(schemaFieldUrn.getEntityKey().get(0));
+    try {
+      Urn resourceUrn = Urn.createFromString(schemaFieldUrn.getEntityKey().get(0));
+      result.setParent(UrnToEntityMapper.map(resourceUrn));
+    } catch (Exception e) {
+      log.error("Error converting schemaField parent urn string to Urn", e);
```

Comment:
ah that's a good catch - `SchemaFieldEntity` is expecting `parent` to not be null here. so I think our options are to either make `parent` nullable or raise an error if this situation occurs (which it shouldn't). I don't think it really makes sense to have `parent` be null since the schemaField should always have a valid parent reference urn in its urn.. so i'm leaning towards raising an error. what do you think?

Reply:
talked IRL and agreed on throwing an exception here